connaisseur/sigstore_validator.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import re
+import subprocess  # nosec
+
+from connaisseur.crypto import decode_and_verify_ecdsa_key
+from connaisseur.exceptions import (
+    CosignError,
+    CosignTimeout,
+    NotFoundException,
+    ValidationError,
+    UnexpectedCosignData,
+)
+
+
+def get_cosign_validated_digests(image: str, pubkey: str):
+    """
+    Gets and processes cosign validation output for a given `image` and `pubkey`
+    and either returns a list of valid digests or raises a suitable exception
+    in case no valid signature is found or cosign fails.
+    """
+    returncode, stdout, stderr = invoke_cosign(image, pubkey)
+    logging.info(
+        "COSIGN output for image: %s; RETURNCODE: %s; STDOUT: %s; STDERR: %s",
+        image,
+        returncode,
+        stdout,
+        stderr,
+    )
+    digests = []
+    if returncode == 0:
+        for sig in stdout.splitlines():
```

Comment:
If stdout.splitlines is empty, you won't catch that here. It will fail later, because then `digests = []`. Dunno, whether that is something you'd want to check for here to have a more specific error or whether you're fine with it failing later due to no digests.

Reply:
I sort of actively decided that the check that there is at least 1 valid digest is handled centrally later to also avoid code duplication. Would keep it, as there seems to be no disadvantage in doing later centrally, or is there?

Reply:
If you do it here, you might get a more detailed error. If `stdout.splitlines()` is empty here but the return code is 0, then there was an error with cosign, which is different from there being no valid digest. If you assume that's not gonna happen, you can also leave it out, but I'd say that would be the more defensive approach of calling another binary (or function in general)